procrastinate/aiopg_connector.py

```diff
@@ -120,30 +166,35 @@ def make_dynamic_query(self, query: str, **identifiers: str) -> str:
             }
         )
 
-    async def wait_for_activity(self):
-        connection = await self._get_connection()
-        try:
-            await asyncio.wait_for(
-                connection.notifies.get(), timeout=self.socket_timeout
-            )
-        except asyncio.TimeoutError:
-            pass
-
-    # This can be called from a signal handler, better not do async stuff
-    def interrupt_wait(self):
-        if not self._connection or self._connection.closed:
-            return
-        asyncio.get_event_loop().call_soon_threadsafe(
-            self._connection.notifies.put_nowait, "s"
-        )
-
-
-class PostgresJobStore(PostgresConnector):
-    def __init__(self, *args, **kwargs):
-        message = (
-            "Use procrastinate.PostgresConnector(...) "
-            "instead of procrastinate.PostgresJobStore(...), with the same arguments"
-        )
-        logger.warn(f"Deprecation Warning: {message}")
-        warnings.warn(DeprecationWarning(message))
-        super().__init__(*args, **kwargs)
+    async def listen_notify(
+        self, event: asyncio.Event, channels: Iterable[str]
+    ) -> NoReturn:
+        # We need to acquire a dedicated connection, and use the listen
+        # query
+        async with self._pool.acquire() as connection:
+            for channel_name in channels:
+                await self._execute_query_connection(
+                    connection=connection,
+                    query=self.make_dynamic_query(
+                        query=sql.queries["listen_queue"], channel_name=channel_name
+                    ),
+                )
+
+            # We'll leave this loop with a CancelError, when we get cancelled
+            while True:
+                event.set()
+                # TODO: because of https://github.com/aio-libs/aiopg/issues/249,
+                # we could get stuck in here forever if the connection closes.
+                # Maybe we should set a timeout and if connection is closed, reopen
+                # a new one.
+                await connection.notifies.get()
```

Comment:
(why did we change the order of event.set() and notifies.get() ? I can't remember)